### PR TITLE
build: add shlint target for shellcheck

### PR DESCRIPTION
**Pull request by janlazo**

### CODE REVIEW

**Review by justinmk — APPROVED** [review]

**Comment by mhinz** [comment]
Funnily enough, the lint job failed because of:

```
In scripts/vim-patch.sh line 504:
  printf "\nInstructions:
         ^-- SC2183: This format string has 3 variables, but is passed 2 arguments.
```

Which is true. The output of `vim-patch -l` was broken (as in the last `%s` was just empty). I fixed that directly on master.

And now, my _local_ shellcheck complains about this:

```
In scripts/vim-patch.sh line 504:
  printf "\nInstructions:
         ^-- SC2183: This format string has 2 variables, but is passed 3 arguments.
```

Which is wrong! It somehow doesn't detect all 3 `%s`. So, I'm using `0.7.0` and apparently Travis uses another version. Maybe it would be a good idea to print the shellcheck version as well.

File: Makefile [hunk 1]
```diff
@@ -138,6 +138,13 @@ functionaltest-lua: | nvim
 lualint: | build/.ran-cmake deps
 	$(BUILD_CMD) -C build lualint
 
+shlint:
+	shellcheck scripts/vim-patch.sh
```
Comment (mhinz) [inline Makefile]:
```suggestion
      shellcheck --version | head -n 2
	shellcheck scripts/vim-patch.sh
```

**Comment by mhinz** [comment]
Also see: https://github.com/neovim/neovim/pull/11351

**Comment by janlazo** [comment]
Travis has shellcheck 0.6. It's in the `system_info` fold.

**Review by mhinz — APPROVED** [review]

**Pull request merged**